dashboard/app/helpers/levels_helper.rb

```diff
@@ -765,9 +765,11 @@ def lab_user_id
   def firebase_auth_token
     return nil unless CDO.firebase_secret
 
+    base_channel = params[:channel_id] || get_channel_for(@level, @user)
     payload = {
       uid: user_or_session_id,
-      is_dashboard_user: !!current_user
+      is_dashboard_user: !!current_user,
+      channel: "#{base_channel}#{CDO.firebase_channel_id_suffix}"
```

Comment:
This looks like it will do the right thing, assuming that any user who is viewing (rather than editing) the project will have this `channel` set.

Reply:
Yes, verified with incognito window that web-shared projects can still write.